incydr/cli/cmds/cases.py

```diff
@@ -101,8 +185,83 @@ def show(ctx: Context, case_number: int, format_: SingleFormat = single_format_o
         echo(case.json())
 
 
-@app.command()
+@cases.command(cls=IncydrCommand)
+@click.argument("case_number")
+@click.option(
+    "--assignee",
+    default=None,
+    help="The administrator assigned to the case. Takes a user ID or a username.  Performs an additional lookup if username is passed.",
+    callback=user_lookup,
+)
+@click.option("--description", default=None, help="Brief optional description.")
+@click.option(
+    "--findings",
+    default=None,
+    help="Markdown formatted text summarizing the findings for a case.",
+)
+@click.option("--name", default=None, help="Case name.")
+@click.option(
+    "--status", default=None, help="Case status. One of `ARCHIVED`, `CLOSED` or `OPEN`."
+)
+@click.option(
+    "--subject",
+    default=None,
+    help="The case subject. Takes a user ID or a username.  Performs an additional lookup if username is passed.",
+    callback=user_lookup,
+)
+@click.pass_context
+def update(
+    ctx: Context,
+    case_number: str,
+    assignee: Optional[str] = None,
+    description: Optional[str] = None,
+    findings: Optional[str] = None,
+    name: Optional[str] = None,
+    status: Optional[str] = None,
+    subject: Optional[str] = None,
+):
+    """
+    Update a single case. Pass the updated value for a field to the corresponding command option.
+    """
+    if not any([assignee, description, findings, name, status, subject]):
+        raise click.UsageError(
+            "At least one command option must be provided to update a case.  Use `cases update --help` to see available options."
+        )
+
+    client = ctx.obj()
+    data = UpdateCaseRequest(
+        assignee=assignee,
+        description=description,
+        findings=findings,
+        name=name,
+        status=status,
+        subject=subject,
+    )
+    client.session.put(f"/v1/cases/{case_number}", json=data.dict())
+
+
+# TODO: do we want to allow usernames for bulk commands, would double the number of lookups
```

Comment:
^

Reply:
I think so, we should make a helper function that caches results so each run we only ever need to lookup a given username once.